test/expect/expire-expire-002.log

```diff
@@ -0,0 +1,1230 @@
+0000000100000000000000FA-0000000000000000000000000000000000000000.gz
+0000000100000000000000FB-0000000000000000000000000000000000000000
+0000000100000000000000FC-0000000000000000000000000000000000000000.gz
+0000000100000000000000FD-0000000000000000000000000000000000000000
+0000000100000000000000FE-0000000000000000000000000000000000000000.gz
+0000000100000000000000FF-0000000000000000000000000000000000000000.gz
+
+[TEST_PATH]/db-master/repo/archive/db/9.3-2/0000000100000001:
+000000010000000100000000-0000000000000000000000000000000000000000
+000000010000000100000001-0000000000000000000000000000000000000000.gz
+000000010000000100000002-0000000000000000000000000000000000000000.gz
+000000010000000100000003-0000000000000000000000000000000000000000
+000000010000000100000004-0000000000000000000000000000000000000000.gz
+000000010000000100000005-0000000000000000000000000000000000000000.gz
+000000010000000100000006-0000000000000000000000000000000000000000
+000000010000000100000007-0000000000000000000000000000000000000000.gz
+000000010000000100000008-0000000000000000000000000000000000000000.gz
+000000010000000100000009-0000000000000000000000000000000000000000
+00000001000000010000000A-0000000000000000000000000000000000000000.gz
+
+Upgrade stanza and expire only earliest db backup and archive
+> [CONTAINER-EXEC] db-master [BACKREST-BIN] --config="[TEST_PATH]/db-master/pgbackrest.conf" --stanza=db --log-level-console=detail --retention-full=3 --retention-archive-type=full expire
+------------------------------------------------------------------------------------------------------------------------------------
+P00   INFO: expire command begin [BACKREST-VERSION]: --config=[TEST_PATH]/db-master/pgbackrest.conf --lock-path=[TEST_PATH]/db-master/repo/lock --log-level-console=detail --log-level-file=trace --log-level-stderr=off --log-path=[TEST_PATH]/db-master/repo/log --repo-path=[TEST_PATH]/db-master/repo --retention-archive-type=full --retention-full=3 --stanza=db
+P00   INFO: expire full backup set: [BACKUP-FULL-1], [BACKUP-INCR-1]
+P00   INFO: remove expired backup [BACKUP-INCR-1]
+P00   INFO: remove expired backup [BACKUP-FULL-1]
+P00 DETAIL: archive retention on backup [BACKUP-FULL-2], archiveId = 9.2-1, start = 00000001000000000000000C
+P00 DETAIL: remove archive: start = 000000010000000000000000, stop = 00000001000000000000000B
+P00 DETAIL: archive retention on backup [BACKUP-FULL-3], archiveId = 9.3-2, start = 000000010000000000000000, stop = 000000010000000000000002
+P00 DETAIL: archive retention on backup [BACKUP-INCR-2], archiveId = 9.3-2, start = 000000010000000000000006, stop = 0000000100000000000000FB
+P00 DETAIL: archive retention on backup [BACKUP-FULL-4], archiveId = 9.3-2, start = 0000000100000000000000FF
+P00 DETAIL: remove archive: start = 000000010000000000000003, stop = 000000010000000000000005
+P00 DETAIL: remove archive: start = 0000000100000000000000FC, stop = 0000000100000000000000FE
```

Comment:
This doesn't look right to me.  If full retention is 3, then all WAL after the oldest unexpired full backup should be retained.  The expiration going on here looks like what would happen if retention-archive were set to 1, but I can't see any indication that is true.